src/Network/WebSockets/Client.hs

```diff
@@ -118,13 +124,31 @@ newClientConnection stream host path opts customHeaders = do
     -- Create the request and send it
     request    <- createRequest protocol bHost bPath False customHeaders
     Stream.write stream (Builder.toLazyByteString $ encodeRequestHead request)
+    checkServerResponse stream request
+    mkConnection stream opts
+  where
+    protocol = defaultProtocol  -- TODO
+    bHost    = T.encodeUtf8 $ T.pack host
+    bPath    = T.encodeUtf8 $ T.pack path
+
+-- | Check the response from the server
+checkServerResponse :: Stream -> RequestHead -> IO ()
+checkServerResponse stream request = do
     mbResponse <- Stream.parse stream decodeResponseHead
     response   <- case mbResponse of
         Just response -> return response
         Nothing       -> throwIO $ OtherHandshakeException $
             "Network.WebSockets.Client.newClientConnection: no handshake " ++
             "response from server"
     void $ either throwIO return $ finishResponse protocol request response
+  where
+    protocol = defaultProtocol -- TODO
+
+
+-- | Create a new Connection from a pre-established stream with already finished
+-- handshake
+mkConnection :: Stream -> ConnectionOptions -> IO Connection
+mkConnection stream opts = do
```

Comment:
Should we rename this to `createConnection` or `streamToConnection`?

Reply:
`createConnection` seems to suffer from the same problem as `mkConnection` (very generic, easy to confuse with `newClientConnection`)
`streamToConnection` or even `streamToClientConnection` would (at least in my mind) make it clearer that it's only wrapping the stream and not performing any handshaking. I'll update the pull request if you agree.

Reply:
Yep, `streamToClientConnection` seems like the best pick.

Reply:
I pushed the change